src/www/views/servers_view/server_connection_indicator/index.ts

```diff
@@ -11,284 +11,191 @@
+
+        --circle-medium-scale: scale(0.66);
+        --circle-medium-delay: ${ANIMATION_DELAY_MS / 2}ms;
+
+        --circle-small-scale: scale(0.33);
+        --circle-small-delay: 0ms;
+
+        --circle-initial-opacity: 0.5;
+        --circle-initial-color: grayscale(0.5);
+
+        --circle-connected-opacity: 1;
+        --circle-connected-color: grayscale(0);
+
+        --circle-disconnected-opacity: 1;
+        --circle-disconnected-color: grayscale(1);
       }
-      .expanded #medium,
-      .expanded #medium-grey {
-        top: 30px;
-        left: 30px;
-        height: 100px;
-        width: 100px;
+
+      :host,
+      .circle {
+        width: 100%;
+        height: 100%;
       }
-      .expanded #large,
-      .expanded #large-grey,
-      .expanded #large-zero {
-        top: 0;
+
+      .circle {
+        position: absolute;
+        display: inline-block;
+
         left: 0;
-        height: 160px;
-        width: 160px;
+
+        transition-property: transform, filter, opacity;
+        transition-duration: var(--duration);
+        transition-timing-function: var(--timing-function);
+
+        animation-duration: var(--duration);
+        animation-timing-function: var(--timing-function);
+        animation-iteration-count: infinite;
       }
-      #large {
-        position: relative;
+
+      .circle-initial {
+        opacity: var(--circle-initial-opacity);
+        filter: var(--circle-initial-color);
       }
-      #large-zero {
-        opacity: 0;
+
+      .circle-connected,
+      .circle-reconnecting,
+      .circle-disconnecting {
+        opacity: var(--circle-connected-opacity);
+        filter: var(--circle-connected-color);
       }
-      #large-zero.INITIAL {
-        opacity: 1;
+
+      .circle-disconnected,
+      .circle-connecting {
+        opacity: var(--circle-disconnected-opacity);
+        filter: var(--circle-disconnected-color);
       }
-      @media (max-width: 360px) {
-        #small,
-        #small-grey {
-          top: 12px;
-          left: 12px;
-          height: 8px;
-          width: 8px;
-        }
-        #medium,
-        #medium-grey {
-          top: 8px;
-          left: 8px;
-          height: 16px;
-          width: 16px;
-        }
-        #large,
-        #large-grey,
-        #large-zero {
-          top: 0;
-          left: 0;
-          height: 32px;
-          width: 32px;
-        }
+
+      .circle-disconnecting {
+        animation-direction: reverse;
+        --timing-function: ease-in;
       }
-      @media (min-height: 600px) {
-        .expanded #small,
-        .expanded #small-grey {
-          top: 72px;
-          left: 72px;
-          height: 48px;
-          width: 48px;
-        }
-        .expanded #medium,
-        .expanded #medium-grey {
-          top: 36px;
-          left: 36px;
-          height: 120px;
-          width: 120px;
+    `,
+    ...CIRCLE_SIZES.map(
+      /* prettier-ignore */
```

Comment:
prettier does NOT like it when i mix the interpolation with the css variables haha